tool/python/singa/layer.py

```diff
@@ -219,14 +219,24 @@ def Feed(self, data, nb_channel=1, is_label=0):
             shapeVector = [batchsize, nb_channel, imgsize, imgsize]
             self.setup(shapeVector)
 
-        data = data.astype(np.float)
-        dataVector = floatVector(datasize)
-        k = 0
-        for i in range(batchsize):
-            for j in range(hdim):
-                dataVector[k] = data[i, j]
-                k += 1
-        self.singalayer.Feed(batchsize, dataVector, is_label)
+        if is_label == 0:
```

Comment:
The signature of the init function is not very general.
nb_channel is specific for image data.
can we use the signature the same as the Feed function,
```
Dummy(shape, data, aux_data)
Label(Dummy, labels)
Image(Dummy, shape, images)
```